lib/components/primitive-components/Port.ts

```diff
@@ -273,8 +275,22 @@ export class Port extends PrimitiveComponent<typeof portProps> {
       } as any) // TODO issue with discriminated union
     }
 
+    const nearestPointInContainer = {
+      x: Math.min(
+        Math.max(portCenter.x, containerCenter.x - containerWidth / 2),
+        containerCenter.x + containerWidth / 2,
+      ),
+      y: Math.min(
+        Math.max(portCenter.y, containerCenter.y - containerHeight / 2),
+        containerCenter.y + containerHeight / 2,
+      ),
+    }
```

Comment:
i'm a bit confused by this logic. Can you break it into a couple variables? It looks like you're computing the `containerTopY`, `containerBottomY` etc. so that could help. But I don't see how this will properly always work intuitively, we could refactor. the code to make it more obvious.

It would also help to have a test that demonstrates that this works, but `localPortInfo.side` is usually defined so i'm not sure how to recreate it, you could break the functionality out into a testable utility function `getDirectionFromPointToNearestSideOfContainer`